tests/pytests/test_client_server_mtls.py

```diff
@@ -0,0 +1,145 @@
+# #####################################################################################
+# Copyright (c) 2021-23, VMware Inc, and the Certifier Authors.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ##############################################################################
+"""
+Basic pytests to exercise stripped-down Client-Server SSL Socket
+communication using mTLS.
+
+# pylint: disable=line-too-long
+
+Sample test program built based on these references:
+
+https://stackoverflow.com/questions/44343230/mutual-ssl-authentication-in-simple-echo-client-server-python-sockets-ssl-m
+https://discuss.python.org/t/ssl-certificate-verify-failed-certificate-verify-failed-ip-address-mismatch-certificate-is-not-valid-for-xxx-xxx-x-xx-ssl-c-997/28403/4
+
+leads to: https://stackoverflow.com/questions/52855924/problems-using-paho-mqtt-client-with-python-3-7
+  where the suggestion to use --addext 'subjectAltName=IP:127.0.0.1' has been provided.
+
+So, finally, generate client / server certificates and private-keys using these commands:
+$ openssl req -new -x509 -days 365 -noenc -out client.pem -keyout client.key --addext 'subjectAltName=IP:127.0.0.1'
+$ openssl req -new -x509 -days 365 -nodes -out server.pem -keyout server.key --addext 'subjectAltName=IP:127.0.0.1'
+# pylint: enable=line-too-long
+"""
+import os
+import socket
+import ssl
+import pprint
+
+# Resolves to current tests/pytests dir
+THIS_SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+
+CERT_HOST = '127.0.0.1'
+CERT_PORT = 1234
+
+# These certificate *.pem and private-key *.key files are generated by
+# the openssl utility.
```

Comment:
If you want to simulate the C++ `sample_app` then `SERVER_CERT` and `CLIENT_CERT` should be signed by the policy certificate

Reply:
Done. Big overhaul of entire test case ...